kvbc/include/categorization/details.h

```diff
@@ -96,5 +97,47 @@ inline void sortAndRemoveDuplicates(std::vector<std::string> &vec) {
   std::sort(vec.begin(), vec.end());
   vec.erase(std::unique(vec.begin(), vec.end()), vec.end());
 }
+// A LocalWriteBatch is used in a multithreaded context, where several jobs need to collect updates.
+// Each job gets its LocalWriteBatch.
+// then all updates are added to the Global NativeWriteBatch, which is not thread safe.
+struct LocalWriteBatch {
+  template <typename KeySpan, typename ValueSpan>
+  void put(const std::string &cFamily, const KeySpan &key, const ValueSpan &value) {
+    puts_.push({cFamily,
+                {std::string(reinterpret_cast<const char *>(key.data()), key.size()),
+                 std::string(reinterpret_cast<const char *>(value.data()), value.size())}});
+    order_.push(put_op_);
+  }
+
+  void moveToBatch(storage::rocksdb::NativeWriteBatch &batch) {
+    while (!order_.empty()) {
+      if (order_.front() == put_op_) {
+        const auto &put = puts_.front();
+        batch.put(put.first, put.second.first, put.second.second);
+        puts_.pop();
+      } else {
+        const auto &del = dels_.front();
+        batch.del(del.first, del.second);
+        dels_.pop();
+      }
+      order_.pop();
+    }
+    ConcordAssert(puts_.empty());
+    ConcordAssert(dels_.empty());
+  }
+
+  template <typename KeySpan>
+  void del(const std::string &cFamily, const KeySpan &key) {
+    dels_.push({cFamily, std::string(reinterpret_cast<const char *>(key.data()), key.size())});
+    order_.push(delete_op_);
+  }
+
+  std::string cFamily_;
+  std::queue<std::pair<std::string, std::pair<std::string, std::string>>> puts_;
```

Comment:
Nit: Might consider using a `struct` instead of pairs for readability, i.e.:

```
struct Put {
    std::string cFamily;
    std::string key;
    std::string value;
};

struct Delete {
    std::string cFamily;
    std::string key;
}
```